ysh/expr_eval.py

```diff
@@ -1564,18 +1552,28 @@ def _EvalSubscript(self, node):
         raise error.InvalidType('expected Dict, List, or Tuple', loc.Missing)
 
     def _EvalAttribute(self, node):
-        # type: (Attribute) -> Any # XXX
-        o = self._EvalExpr(node.obj)
+        # type: (Attribute) -> value_t
+        obj = _PyObjToValue(self._EvalExpr(node.obj))
+        UP_obj = obj
         id_ = node.op.id
         if id_ == Id.Expr_RArrow:
             # Used for s->startswith(x)
-            name = node.attr.tval
-            return getattr(o, name)
+            func_name = node.attr.tval
+            func = self.LookupVar(func_name, node.attr)
+            # If there are collisions on method names for different types, mem
```

Comment:
Yeah this is probably a good idea

we may need to change it for user-defined functions/methods too